config/initializers/mail_observers.rb

```diff
@@ -2,7 +2,9 @@
 # otherwise the observer won't be invoked.
 #
 require_relative "../../app/services/email_delivery_observer"
+require_relative "../../app/services/email_delivering_interceptor"
 
 ActiveSupport.on_load(:action_mailer) do |mailer|
   mailer.register_observer EmailDeliveryObserver
+  mailer.register_observer EmailDeliveringInterceptor
```

Comment:
```suggestion
  mailer.register_interceptor EmailDeliveringInterceptor
``` 

d'ailleurs tu pourrais intervertir les 2 vu que l'interceptor est appelé + tot